File: app-android/app/src/main/java/com/health2u/data/local/entity/ExamEntity.kt
package com.health2u.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.Index

@Entity(
    tableName = "exams",
    indices = [Index(value = ["userId"]), Index(value = ["date"])]
)
data class ExamEntity(
    @PrimaryKey val id: String,
    val userId: String,
    val title: String,
    val type: String,
    val date: Long,
    val fileUrl: String?,
    val localFilePath: String?,
    val notes: String?,
    val createdAt: Long,
    val updatedAt: Long
)

File: app-ios/Sources/Health2u/Data/Local/ExamEntity.swift
import Foundation

public struct ExamEntity: Sendable, Equatable, Identifiable {
    public let id: String
    public var userId: String
    public var title: String
    public var type: String
    public var date: Date
    public var fileUrl: String?
    public var localFilePath: String?
    public var notes: String?
    public var createdAt: Date
    public var updatedAt: Date
    public var analysis: ExamAnalysisEntity?

    public init(
        id: String,
        userId: String,
        title: String,
        type: String,
        date: Date,
        fileUrl: String? = nil,
        localFilePath: String? = nil,
        notes: String? = nil,
        createdAt: Date,
        updatedAt: Date,
        analysis: ExamAnalysisEntity? = nil
    ) {
        self.id = id
        self.userId = userId
        self.title = title
        self.type = type
        self.date = date
        self.fileUrl = fileUrl
        self.localFilePath = localFilePath
        self.notes = notes
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.analysis = analysis
    }
}
